py-rattler/rattler/repo_data/package_record.py

```diff
@@ -10,19 +14,220 @@ class PackageRecord:
     of a package on a Conda channel.
     """
 
-    def __init__(self) -> None:
-        self._package_record = PyPackageRecord()
+    _record: PyRecord
 
-    @classmethod
-    def _from_py_package_record(
-        cls, py_package_record: PyPackageRecord
+    @staticmethod
+    def from_index_json(
+        path: os.PathLike[str],
+        size: Optional[int] = None,
+        sha256: Optional[str] = None,
+        md5: Optional[str] = None,
     ) -> PackageRecord:
+        """
+        Builds a PackageRecord from a IndexJson.
+
+        Examples
+        --------
+        ```python
+        >>> record = PackageRecord.from_index_json(
+        ...     "../test-data/conda-meta/pysocks-1.7.1-pyh0701188_6.json"
+        ... )
+        >>> type(record)
+        <class 'rattler.repo_data.package_record.PackageRecord'>
+        >>>
+        ```
+        """
+        return PackageRecord._from_py_record(
+            PyRecord.from_index_json(path, size, sha256, md5)
+        )
+
+    @staticmethod
+    def sort_topologically(records: List[PackageRecord]) -> List[PackageRecord]:
+        """
+        Sorts the records topologically.
+        This function is deterministic, meaning that it will return the same result
+        regardless of the order of records and of the depends vector inside the records.
+        Note that this function only works for packages with unique names.
+
+        Examples
+        --------
+        ```python
+        >>> from os import listdir
+        >>> from os.path import isfile, join
+        >>> records = [
+        ...     PackageRecord.from_index_json(join("../test-data/conda-meta/", f))
```

Comment:
These `conda-meta` files are not actually `index.json` files. Would it be better to simply load them as PrefixRecords? 

```suggestion
        ...     PrefixRecord.from_path(join("../test-data/conda-meta/", f))
```

Reply:
Yeah this would work too, lemme do this instead